common/lib/dependabot/pull_request_updater/github.rb

```diff
@@ -163,7 +163,8 @@ def update_branch(commit)
         return nil if e.message.match?(/Reference cannot be updated/i)
 
         if e.message.match?(/force\-push to a protected/i) ||
-           e.message.match?(/not authorized to push/i)
+           e.message.match?(/not authorized to push/i) ||
+           e.message.match?(/must not contain merge commits/)
```

Comment:
Since this case is not well understood, it makes sense to treat it as a protected branch error, but I wonder if we should instrument this specific outcome.

I guess the question is if this is the result of Dependabot, or a user modifying a Dependabot branch somehow?

Reply:
I think this will be a user change enforcing this branch protection rule on all branches, e.g. using `*`

Reply:
Could raise another error and instrument in the backend but inclined to leave it for now as it seems to be so niche

Reply:
Good call!